pkgs/applications/terminal-emulators/wezterm/default.nix

```diff
@@ -34,24 +36,15 @@ rustPlatform.buildRustPackage rec {
     owner = "wez";
     repo = "wezterm";
     rev = version;
-    fetchSubmodules = true;
-    hash = "sha256-Az+HlnK/lRJpUSGm5UKyma1l2PaBKNCGFiaYnLECMX8=";
+    hash = "sha256-hhuzs89wWKc7n6HMKriWvV+pLhfLvO06XRWtmdCQ0rs=";
   };
 
-  patches = [
-    # Remove unused `fdopen` in vendored zlib, which causes compilation failures with clang 19 on Darwin.
-    # Ref: https://github.com/madler/zlib/commit/4bd9a71f3539b5ce47f0c67ab5e01f3196dc8ef9
-    ./zlib-fdopen.patch
-
-    # Fix platform check in vendored libpng with clang 19 on Darwin.
-    (fetchpatch2 {
-      url = "https://github.com/pnggroup/libpng/commit/893b8113f04d408cc6177c6de19c9889a48faa24.patch?full_index=1";
-      extraPrefix = "deps/freetype/libpng/";
-      stripLen = 1;
-      excludes = [ "deps/freetype/libpng/AUTHORS" ];
-      hash = "sha256-zW/oUo2EGcnsxAfbbbhTKGui/lwCqovyrvUnylfRQzc=";
-    })
-  ];
+  prePatch = ''
+    tar -xf ${freetype.src} --strip-components=1 -C deps/freetype/freetype2
+    tar -xf ${libpng.src} --strip-components=1 -C deps/freetype/libpng
+    tar -xf ${zlib.src} --strip-components=1 -C deps/freetype/zlib
+    tar -xf ${harfbuzz.src} --strip-components=1 -C deps/harfbuzz/harfbuzz
```

Comment:
Not sure if we should do that when there are files that are generated against the source code https://github.com/wez/wezterm/blob/main/deps/harfbuzz/src/lib.rs

Reply:
Upstream just updated all libraries to their last version when I asked them, so it seems fine.